### ISSUE TITLE
counsel-git-grep won't accept negation group

### ISSUE THREAD

**Issue by drym3r** [abo-abo/swiper#867]
Imagine I have a file on a git repository with this content:
```
Hi github
clone this git repository
```
When I open the file, I execute swiper and I write "git ! github" it will show me the second line. But if I execute counsel-git-grep, when I write the same it won't return anything.
My ivy-re-builders-alist is set:
```
(setq ivy-re-builders-alist
  '((read-file-name-internal . ivy--regex-fuzzy)
    (t . ivy--regex-plus)))
```
Tell me if I can help debugging.

**Comment by abo-abo** [comment]
It's not an issue of debugging, the feature simply isn't implemented yet. Negation works for sync completion, but not for async completion.
The reason is that sync matching happens fully inside Emacs, while async matching happens in an external process. It's possible to add negation to async, I'll try to do it at some point.

**Comment by drym3r** [comment]
I see. Then is not possible to negate at all or I should use the program's syntax?

**Comment by abo-abo** [comment]
> Then is not possible to negate at all or I should use the program's syntax?

It should be possible to use the program's syntax to negate.

I'm not aware if `git-grep` accepts any negation switches though. It looks like you're expected to use the pipe:

    git grep swiper | grep -v point

**Comment by drym3r** [comment]
Yeah, I don't find any other way to negate than piping, how unfortunate. It would be cool to use the ! syntax.

**Comment by basil-conto** [comment]
> I'm not aware if `git-grep` accepts any negation switches though.

It does, via `--not`. Stay tuned :).
